tests/Output/Integration/StrictUnifiedDiffOutputBuilderIntegrationTest.php

```diff
@@ -0,0 +1,314 @@
+
+        $output = $p->getOutput();
+
+        $diffLines    = \preg_split('/(.*\R)/', $diff, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY);
+        $diffLines[0] = \preg_replace('#^\-\-\- .*#', '--- /' . $this->fileFrom, $diffLines[0], 1);
+        $diffLines[1] = \preg_replace('#^\+\+\+ .*#', '+++ /' . $this->fileFrom, $diffLines[1], 1);
+        $diff         = \implode('', $diffLines);
+
+        $outputLines    = \preg_split('/(.*\R)/', $output, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY);
+        $outputLines[0] = \preg_replace('#^\-\-\- .*#', '--- /' . $this->fileFrom, $outputLines[0], 1);
+        $outputLines[1] = \preg_replace('#^\+\+\+ .*#', '+++ /' . $this->fileFrom, $outputLines[1], 1);
+        $output         = \implode('', $outputLines);
+
+        $this->assertSame($diff, $output);
+    }
+
+    private function doIntegrationTestGitApply(string $diff, string $from, string $to)
+    {
+        $this->assertNotSame('', $diff);
+        $this->assertValidUnifiedDiffFormat($diff);
+
+        $diff = self::setDiffFileHeader($diff, $this->fileFrom);
+
+        $this->assertNotFalse(\file_put_contents($this->fileFrom, $from));
+        $this->assertNotFalse(\file_put_contents($this->filePatch, $diff));
+
+        $p = new Process(\sprintf(
+            'git --git-dir %s apply --check -v --unsafe-paths --ignore-whitespace %s',
+            \escapeshellarg($this->dir),
+            \escapeshellarg($this->filePatch)
+        ));
+
+        $p->run();
+
+        $this->assertProcessSuccessful($p);
+    }
+
+    private function doIntegrationTestPatch(string $diff, string $from, string $to)
+    {
+        $this->assertNotSame('', $diff);
+        $this->assertValidUnifiedDiffFormat($diff);
+
+        $diff = self::setDiffFileHeader($diff, $this->fileFrom);
+
+        $this->assertNotFalse(\file_put_contents($this->fileFrom, $from));
+        $this->assertNotFalse(\file_put_contents($this->filePatch, $diff));
+
+        $command = \sprintf(
+            'patch -u --verbose --posix %s < %s',
+            \escapeshellarg($this->fileFrom),
+            \escapeshellarg($this->filePatch)
+        );
+
+        $p = new Process($command);
+        $p->run();
+
+        $this->assertProcessSuccessful($p);
+
+        $this->assertStringEqualsFile(
+            $this->fileFrom,
+            $to,
+            \sprintf('Patch command "%s".', $command)
+        );
+    }
+
+    protected function tearDown()
+    {
+        $this->cleanUpTempFiles();
+    }
+
+    private function assertProcessSuccessful(Process $p)
+    {
+        $this->assertTrue(
+            $p->isSuccessful(),
+            \sprintf(
+                "Command exec. was not successful:\n\"%s\"\nOutput:\n\"%s\"\nStdErr:\n\"%s\"\nExit code %d.\n",
+                $p->getCommandLine(),
+                $p->getOutput(),
+                $p->getErrorOutput(),
+                $p->getExitCode()
+            )
+        );
+    }
+
+    private function cleanUpTempFiles()
```

Comment:
(not only in this test)
why `cleanUpTempFiles` is called in both, `setUp` and `tearDown` ?

Reply:
The overhead is zero to none and it saves a lot of time when debugging, especially if you kill the test during a step-through to inspect the generated files. 
When resume after that it is nice the test cleans up the generated file before starting, creating a new clean playing field.